crates/rpc/eth/block.rs

```diff
@@ -400,7 +400,13 @@ pub fn get_transaction_by_hash(
             Some(transaction) => transaction,
             _ => return Ok(Value::Null),
         };
-
+    let (block_number, index) = storage
+        .get_transaction_location(request.transaction_hash)
+        .unwrap()
+        .unwrap();
```

Comment:
It would be better to use `?` for the Result and match the Option to prevent failing on None case (like the rest of the rpc methods)